settings.php

```diff
@@ -24,7 +24,7 @@
 
 defined('MOODLE_INTERNAL') || die;
 
-if ($ADMIN->fulltree) {
+if ($hassiteconfig && is_enabled_auth('basic')) {
```

Comment:
Why did you modify this? You generally want to be able to configure the settings for an auth plugin before you turn it on, and to freely enable or disable it later